boot/freeldr/freeldr/lib/fs/ntfs.c

```diff
@@ -417,23 +510,23 @@ static PNTFS_ATTR_CONTEXT NtfsFindAttributeHelper(PNTFS_VOLUME_INFO Volume, PNTF
             }
         }
 
-        if (AttrRecord->Type == Type)
+        if (AttrRecord->Type == Type &&
+            AttrRecord->NameLength == NameLength &&
+            NtfsGetAttributeSize(AttrRecord) != 0 && // HACK for ntfs3 driver on linux!!!
+            (Instance == (ULONG)-1 || (ULONG)AttrRecord->Instance == Instance))
```

Comment:
Maybe then change the type of the `Instance` function parameter to USHORT , so that the cast wouldn't be necessary?